cryostatio/cryostat3
# feat(eventtemplates): custom event templates in S3

# Welcome to Cryostat3! 👋
## Before contributing, make sure you have:

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes #13
Replaces #119
Related to #42

## Description of the change:
This implements custom event template storage in S3. This works much the same as custom event templates do in 2.4.0, but rather than stored in a local filesystem location the templates are stored in and retrieved from S3.

## How to manually test:
1. *Run CRYOSTAT_IMAGE=quay.io... bash smoketest.bash -Ot*
2. Go to Events
3. Select a target
4. Download an event template (other than `ALL`) - you should find ex. `Profiling.jfc` downloaded to your workstation. This step requires https://github.com/cryostatio/cryostat-web/pull/1215 . Without it, you can check your workstation's JVM installation and look for ex. `/usr/lib/jvm/java/lib/jfr/profile.jfc` and use that as a base, instead: `cp /usr/lib/jvm/java/lib/jfr/profile.jfc ~/Downloads/Profiling.jfc`.
5. Open `Profiling.jfc` with a text or XML editor, or the JMC event configuration wizard. Make some changes to the event template file, including changing the `label`, `description`, and `provider` attributes at the top.
6. Back in the Cryostat UI, upload your modified event template back to Cryostat.
7. Create a new Recording using your new event template.
8. Delete the event template. ~~This will display a notification but the UI table may not update - I think this is a -web bug. Navigating away and back to Events will show the correctly updated contents.~~
